collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -129,15 +121,21 @@ protected void attachBaseContext(Context base) {
     @Override
     public void onCreate() {
         super.onCreate();
-        ExternalFilesUtils.testExternalFilesAccess(this);
-
         singleton = this;
-        setupDagger();
-        DaggerUtils.getComponent(this).inject(this);
 
-        applicationInitializer.initialize();
-        fixGoogleBug154855417();
-        setupStrictMode();
+        CrashHandler crashHandler = CrashHandler.install(this);
+        boolean shouldLaunch = crashHandler.checkConditions(() -> {
```

Comment:
Could be:
`        boolean shouldLaunch = crashHandler.checkConditions(() -> ExternalFilesUtils.testExternalFilesAccess(this));
`

Reply:
and it doesn't even need to return any value. It could have another `Runnable` as a parameter to run setup if everything is fine. Like:
```
        crashHandler.checkConditions(() -> {
            ExternalFilesUtils.testExternalFilesAccess(this);
        }, () -> {
            setupDagger();
            DaggerUtils.getComponent(this).inject(this);

            applicationComponent.applicationInitializer().initialize();
            fixGoogleBug154855417();
            setupStrictMode();
        });
```
Probably we would need to set the name of that method then too.

Reply:
I went back and forth on versions like this, but couldn't decide what was best. Will go with the version you suggested.